src/HW_models/NRF_RTC.c

```diff
@@ -125,15 +124,52 @@ static void time_of_1_counter_wrap(int rtc, bs_time_t *us, uint16_t *frac) {
   *us = counter_to_time((uint64_t)RTC_COUNTER_MASK + 1, rtc, frac);
 }
 
+/**
+ * Convert a counter value to absolute time.
+ * @note To obtain a time delta, use @ref counter_to_time instead
+ */
+static bs_time_t counter_to_abs_time(int rtc, uint32_t rtc_counter) {
+  uint16_t frac;
+  bs_time_t bs_time = RTC_counter_startT[rtc]
+                      + counter_to_time(rtc_counter, rtc, &frac);
+
+  bs_time_t now = tm_get_hw_time();
+
+  if (bs_time <= now) {
+    bs_time_t t_us;
+    uint16_t t_frac;
+
+    time_of_1_counter_wrap(rtc, &t_us, &t_frac);
+
+    do {
+      bs_time += t_us;
+      frac += t_frac;
+      if (frac >= 0x200){
+        frac -= 0x200;
+        bs_time +=1;
+      }
+    } while (bs_time <= now);
+  }
+
+  return bs_time;
+}
+
 static void update_master_timer() {
   Timer_RTC = TIME_NEVER;
   for (int rtc = 0; rtc < N_RTC ; rtc++) {
     if (RTC_Running[rtc] == false) {
       continue;
     }
-    for (int cc = 0 ; cc < N_CC ; cc++) {
-      if (cc_timers[rtc][cc] < Timer_RTC) {
-        Timer_RTC = cc_timers[rtc][cc];
+
+    if (NRF_RTC_regs[rtc].EVTEN & RTC_INTENSET_TICK_Msk) {
```

Comment:
This is not correct. Just because *some* RTC has reached its count, it does not mean *this* RTC has.
Similarly, when the event or interrupt is enabled, you need to ensure Timer_RTC is updated to be at least the next time that RTC will count.
You could keep a separate array of timers (for ex. cc_timers_ticks[rtc]) which you update :
* as a sideeffect of enabling the interupt 
* as a sideeffect of enabling the event
* each time that particular counter counts one tick (if ticks are enabled) (by moving it one count period ahead)

Reply:
Yes, you are right. It is fixed now